src/recording.rs

```diff
@@ -12,11 +12,11 @@ pub struct ResourceId(pub NonZeroU64);
 
 impl ResourceId {
     pub fn next() -> ResourceId {
-        static ID_COUNTER: AtomicU64 = AtomicU64::new(0);
+        static ID_COUNTER: AtomicU64 = AtomicU64::new(1);
```

Comment:
```suggestion
        // SAFETY: We initialize with 1 because unsafe code expects this to be non-zero.
        static ID_COUNTER: AtomicU64 = AtomicU64::new(1);
```